opentelemetry/src/global/internal_logging.rs

```diff
@@ -0,0 +1,200 @@
+#![allow(unused_macros)]
```

Comment:
Why expose this from the `opentelemetry` (API) package if it is seemingly only going to be used by SDK or implementation specific crates in the project? Won't changes to this stuff necessitate a possible semver change in `opentelemetry` API crate itself, unnecessarily?

Reply:
I kept it at the level of custom error handler which is within API. I thought there could be the cases where we may want to used within API e.g., debugging `Context` implementation.

Reply:
i am not sure if there are better ways to "hide" it. It should be public API, so it follows the same sem ver rules, and it can be used by OTel itself + any other components extending otel. For example, if someone authors an exporter, they can also use this. 
We can doc it so user end up misusing it for any other logging!

> Won't changes to this stuff necessitate a possible semver change in opentelemetry API crate itself, unnecessarily?
We should treat this API as just any other API, so it cannot make breaking changes. As long as we are doing a minimal API (super thin wrapper around tracing), we should be able to stabilize it by 1.0, and keep it non-breaking forever (or until 2.0 ships)

Reply:
@shaun-cox The current macros are lightweight and closely resemble the tracing macros. Since not all opentelemetry-* crates depend on opentelemetry-sdk, exposing these macros from the API package appears to be the most suitable approach. 

Let me know if you see any concerns here.